pynest/nest/lib/hl_api_types.py

```diff
@@ -880,3 +883,51 @@ def apply(self, spatial_nc, positions=None):
                 if len(pos) != len(positions[0]):
                     raise ValueError('All positions must have the same number of dimensions')
             return sli_func('Apply', self._datum, {'source': spatial_nc, 'targets': positions})
+
+
+def serializable(data):
+    """Make data serializable for JSON.
+
+    Parameters
+    ----------
+    data : any
+
+    Returns
+    -------
+    data_serialized : str, int, float, list, dict
+        Data can be encoded to JSON
+    """
+
+    if isinstance(data, (numpy.ndarray, NodeCollection)):
+        data_serialized = data.tolist()
+    elif isinstance(data, SynapseCollection):
+        # Get full information from SynapseCollection
+        data_serialized = serializable(data.get())
+    elif isinstance(data, kernel.SLILiteral):
+        # Get name of SLILiteral.
+        data_serialized = data.name
+    elif isinstance(data, (list, tuple)):
+        data_serialized = [serializable(d) for d in data]
+    elif isinstance(data, dict):
+        data_serialized = dict([(key, serializable(value)) for key, value in data.items()])
+    else:
+        data_serialized = data
+    return data_serialized
+
+
+def to_json(data):
+    """Serialize data to JSON.
+
+    Parameters
+    ----------
+    data : any
+
+    Returns
+    -------
+    data_json : str
+        JSON format of the data
+    """
+
+    data_serialized = serializable(data)
+    data_json = json.dumps(data_serialized)
```

Comment:
Will a human ever look at the generated JSON? Possibly insert a ``indent=4`` to enable pretty-printing.

Reply:
I added keyword argument passing to json.dumps. You can use it by nest.to_json(data, indent=4). Is it acceptable?